backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -251,7 +251,7 @@ public Building Update(Building building)
             foreach (var parcel in building.Parcels.Select(pb => pb.Parcel))
             {
                 // Check if the parcel already exists.
-                var existingAssociatedParcel = existingBuilding.Parcels
+                var existingAssociatedParcel = existingBuilding.Parcels.Where(pb => pb.Parcel.Id != 0)
```

Comment:
in the case of multiple new parcels, this logic was erroneously allowing pending parcels that have yet to be added to be treated as existing.